bagit.py

```diff
@@ -182,34 +214,73 @@ def make_bag(bag_dir, bag_info=None, processes=1, checksums=None, checksum=None,
         else:
             LOGGER.info(_("Creating data directory"))
 
-            # FIXME: if we calculate full paths we won't need to deal with changing directories
-            os.chdir(bag_dir)
-            cwd = os.getcwd()
-            temp_data = tempfile.mkdtemp(dir=cwd)
+            data_subdir = None
+            if dest_dir:
+                temp_data = tempfile.mkdtemp(dir=dest_dir)
+                data_dir = os.path.join(dest_dir, 'data')
+                data_subdir = os.path.relpath(
+                    bag_dir, os.path.join(bag_dir, os.pardir)
+                )
+                if os.path.abspath(os.path.join(bag_dir, data_subdir)) == bag_dir:
```

Comment:
I think this could be `os.path.samefile(os.path.join(bag_dir, data_subdir), bag_dir)`